src/snackbar.jsx

```diff
@@ -276,14 +276,18 @@ const Snackbar = React.createClass({
     );
 
     return (
-      <div {...others} style={prepareStyles(Object.assign(styles.root, style))}>
+      <ClickAwayListener
+        {...others}
+        onClickAway={open && this.componentClickAway}
```

Comment:
The `ClickAwayListener` doesn't create listeners if you don't provide an `onClickAway` callback. So listeners will only be registered when the snackbar is open.